src/advanced_inv.cpp

```diff
@@ -1752,15 +1753,18 @@ bool advanced_inventory::remove_item( advanced_inv_listitem &sitem )
         assert( &cont->contents.front() == sitem.it );
         cont->contents.erase( cont->contents.begin() );
         rc = true;
-    } else if( s.can_store_in_vehicle() && panes[src].in_vehicle() ) {
-        rc = s.veh->remove_item( s.vstor, sitem.it );
     } else if( sitem.area == AIM_WORN ) {
         rc = g->u.takeoff( sitem.it );
-    } else {
+    } else if( s.can_store_in_vehicle() ) {
+        rc = s.veh->remove_item( s.vstor, sitem.it );
+    }
+
+    if( !rc ) {
         g->m.i_rem( s.pos, sitem.it );
-        rc = true;
+        rc = true; // NOT ALWAYS! FIX!
     }
-    return rc;
+
+    return rc; // Always returns true
```

Comment:
I'm marking it as a "FIX THIS CRAP LATER, IT'S UGLY" quick fix hack for now, later on the `map::i_rem` will probably be changed to return bool, like `vehicle::remove_item`.

It is intentionally "wrong".